sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -155,13 +159,20 @@ export type RemoveLiquidityOptions = CommonOptions & RemoveLiquiditySpecificOpti
 export type CollectOptions = CommonOptions & CollectSpecificOptions
 
 // type guard
-// @ts-ignore
 function isMint(options: AddLiquidityOptions): options is MintOptions {
   return Object.keys(options).some((k) => k === 'recipient')
 }
 
+function shouldCreatePool(options: MintOptions): boolean {
+  if (options.createPool) {
+    invariant(options.sqrtPriceX96 !== undefined, 'NO_SQRT_PRICE')
```

Comment:
Hm do you prefer it the other way? I feel like this makes more sense to me because the pool on the position object shouldnt exist if createPool = true. Like its passed in because you're now initializing

Reply:
i mean it could exist but I feel like its more error prone bc in all other cases you should initialize a position with a pool that has all current up to date values

Reply:
Yep its fine with me; was just calling out for my own sake